### PR TITLE
Add feedback on custom action

### PR DESCRIPTION
**Pull request by GrimMaple**
When a custom action is completed it will now report back to the user.

I also made it so pressing the action button will not hang the app.

### CODE REVIEW

File: mrHelper/src/mrHelperForm.cs [hunk 1]
```diff
@@ -97,6 +102,15 @@ private void addCustomActions()
          }
       }
 
+      private void setCustomActionsEnabled(bool enabled)
+      {
+         foreach (var control in groupBoxActions.Controls)
+         {
+            if (control is Button)
```
Comment (denis-adamchuk) [inline mrHelper/src/mrHelperForm.cs]: do we really need this restriction?

File: mrHelper/src/mrHelperForm.cs [hunk 2]
```diff
@@ -80,11 +81,15 @@ private void addCustomActions()
...
             };
-            button.Click += (x, y) =>
+            button.Click += async (x, y) =>
             {
                try
                {
-                  command.Run();
+                  setCustomActionsEnabled(false);
+                  await Task.Run(() => command.Run());
+                  MessageBox.Show("Custom action successfully completed", "Info",
```
Comment (denis-adamchuk) [inline mrHelper/src/mrHelperForm.cs]: I think we need something less annoying that message box. I thought about tooltips or status bar update. Something that do not need *one more* user actions.

File: mrHelper/src/mrHelperForm.cs [hunk 3]
```diff
@@ -80,11 +81,15 @@ private void addCustomActions()
                Enabled = false,
                TabStop = false
             };
-            button.Click += (x, y) =>
+            button.Click += async (x, y) =>
```
Comment (denis-adamchuk) [inline mrHelper/src/mrHelperForm.cs]: btw, we do we need async here? why ordinary lambda was not enough?

Comment (GrimMaple) [reply mrHelper/src/mrHelperForm.cs]: The ordinary lambda would block the execution until the action is complete. `async` event callback will not block, so the user can continue working while the action is still performing its job.

File: mrHelper/src/mrHelperForm.cs (hunk 2 above)
Comment (GrimMaple) [reply mrHelper/src/mrHelperForm.cs]: How about a color or text change for a brief moment?
For example, after the action is successfully completed, the text on the button becomes "Success!" for a few seconds and then changes back to normal?

File: mrHelper/src/mrHelperForm.cs (hunk 1 above)
Comment (GrimMaple) [reply mrHelper/src/mrHelperForm.cs]: It is certainly helpful to prevent NullPointerReference exceptions

File: mrHelper/src/mrHelperForm.cs (hunk 3 above)
Comment (denis-adamchuk) [reply mrHelper/src/mrHelperForm.cs]: ok

File: mrHelper/src/mrHelperForm.cs (hunk 2 above)
Comment (denis-adamchuk) [reply mrHelper/src/mrHelperForm.cs]: Can you please try and make sure it does not look weird? You know, may be we need a status bar to reflect these and other user actions? For example, when git fetch/clone is running or MR list is loading. I just want that UI look consistent and not very different.

File: mrHelper/src/mrHelperForm.cs (hunk 1 above)
Comment (denis-adamchuk) [reply mrHelper/src/mrHelperForm.cs]: Winforms base class Control has Enabled property, why don't check it? If you believe there could be null pointers, let's check for 'null'.

**Comment by denis-adamchuk** [comment]
it is very outdated.

**Pull request closed without merging**